components/proxy/src/main/java/com/hotels/styx/admin/handlers/MetricsHandler.java

```diff
@@ -39,6 +61,61 @@
      * @param cacheExpiration duration for which generated page content should be cached
      */
     public MetricsHandler(CodaHaleMetricRegistry metricRegistry, Optional<Duration> cacheExpiration) {
-        super(checkNotNull(metricRegistry.getMetricRegistry()), cacheExpiration, new MetricsModule(SECONDS, MILLISECONDS, DO_NOT_SHOW_SAMPLES));
+        super(requireNonNull(metricRegistry.getMetricRegistry()), cacheExpiration, new MetricsModule(SECONDS, MILLISECONDS, DO_NOT_SHOW_SAMPLES));
+
+        this.metricRegistry = metricRegistry;
+    }
+
+    @Override
+    public Observable<HttpResponse> handle(HttpRequest request) {
+        return metricName(request.path())
+                .map(metricName -> specificMetricsResponse(request, metricName))
+                .orElseGet(() -> super.handle(request));
+    }
+
+    private Observable<HttpResponse> specificMetricsResponse(HttpRequest request, String metricName) {
+        Map<String, Metric> metrics = metrics(metricName);
+
+        if (metrics.isEmpty()) {
+            return just(response(NOT_FOUND).build().toStreamingResponse());
+        }
+
+        boolean pretty = request.queryParam("pretty").isPresent();
```

Comment:
I am starting to think this "pretty" query param should be moved out into a separate class, and somehow composed with other admin handlers. But that would be a separate improvement.